### ISSUE TITLE
multi/exec has unexpected result

### ISSUE THREAD

**Issue by mocheng** [NodeRedis/node_redis#624]
As of below code 

```
var redis = require('redis');
var redisClient = redis.createClient(6379, 'localhost');

var oldKey = 'my';
var newKey = 'my';

redisClient.multi()
  .rename(oldKey, newKey)
  .exec(function(err, results) {
    console.log(err);
    console.log(results);
  });
});
```
According to doc https://github.com/mranney/node_redis#user-content-multiexec-callback- , the `exec` callback will have error in `err` arguments and results in `results` argument.

I expect it to output
>[ 'ERR source and destination objects are the same' ]
undefined

Actually, it outputs
>null
[ 'ERR source and destination objects are the same' ]

Without `multi` and `exec`, 
```
  redisClient.rename(oldKey, newKey, function(err, result) {
    console.log(err);
    console.log(result);
  });
```

output: 

>[Error: ERR source and destination objects are the same]
undefined

It is clear that `err` exists but `result` is empty.

It seems that in multi mode,  `err` and `results` are getting confused.

**Comment by blainsmith** [comment]
In order to better support this project and its new group of collaborators under a new org we are trying to clean up and close issues older than 6 months. Your issue may have been fixed since the issue was open, however, if you are still experiencing a problem **please** re-open this issue and we will label it accordingly.

**Comment by BridgeAR** [comment]
This will be fixed by #826
